cmake/FindOpenBLAS.cmake

```diff
@@ -64,15 +81,27 @@ if(OpenBLAS_FOUND)
     message(STATUS "Found OpenBLAS libraries: ${OpenBLAS_LAPACK_LIB}")
     message(STATUS "Found OpenBLAS include: ${OpenBLAS_INCLUDE_DIR}")
   endif()
+  get_filename_component(OpenBLAS_LIB_DIR ${OpenBLAS_LIB} PATH)
+  vp_get_version_from_pkg2(blas-openblas "${OpenBLAS_LIB_DIR}/pkgconfig" OpenBLAS_VERSION)
```

Comment:
Option to parse library version seems to be `vp_parse_header`, `vp_parse_header2` and `vp_get_version_from_pkg`.

---

On Ubuntu 16.04 with `0.2.18-1ubuntu1`, in the content of `/usr/include/openblas/openblas_config.h` there is:

```
#define OPENBLAS_VERSION " OpenBLAS 0.2.18 "
```

Content of `/usr/lib/pkgconfig/blas-openblas.pc` (notice `Version: 0.2.18`):

```
prefix=/usr
libdir=${prefix}/lib/openblas-base
includedir=${prefix}/include/openblas
Name: openblas-blas
Description: Optimized BLAS (linear algebra) library based on GotoBLAS2
Version: 0.2.18
URL: https://github.com/xianyi/OpenBLAS
Libs: -L${libdir} -lblas
Libs.private: -L${libdir} -lblas -lgfortran -lpthread -lm
Cflags: -I${includedir}
```

---

On Ubuntu 18.04, in the content of `/usr/include/x86_64-linux-gnu/openblas_config.h` there is:

```
#define OPENBLAS_VERSION " OpenBLAS 0.2.20 "
```

Content of `/usr/lib/x86_64-linux-gnu/pkgconfig/openblas.pc` (notice `version=` and `Version: `):

```
libdir=/usr/lib/x86_64-linux-gnu
includedir=/usr/include/x86_64-linux-gnu
version=0.2.20
extralib=-lm -lpthread -lgfortran -lm -lpthread -lgfortran
Name: openblas
Description: OpenBLAS is an optimized BLAS library based on GotoBLAS2 1.13 BSD version
Version: ${version}
URL: https://github.com/xianyi/OpenBLAS
Libs: -L${libdir} -lopenblas
Libs.private: ${extralib}
Cflags: -I${includedir}
```

---

On Ubuntu 20.04, in the content of `/usr/include/x86_64-linux-gnu/openblas_config.h` there is:

```
#define OPENBLAS_VERSION " OpenBLAS 0.3.8 "
```

Content of `/usr/lib/x86_64-linux-gnu/pkgconfig/openblas.pc`:

```
libdir=/usr/lib/x86_64-linux-gnu/openblas-pthread/
includedir=/usr/include/x86_64-linux-gnu/openblas-pthread/
openblas_config= USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS= NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
version=0.3.8
extralib=-lm -lpthread -lgfortran -lm -lpthread -lgfortran
Name: openblas
Description: OpenBLAS is an optimized BLAS library based on GotoBLAS2 1.13 BSD version
Version: ${version}
URL: https://github.com/xianyi/OpenBLAS
Libs: -L${libdir} -lopenblas
Libs.private: ${extralib}
Cflags: -I${includedir}
```

---

When building from source, `openblas_config.h`:

```
#define OPENBLAS_VERSION " OpenBLAS 0.3.12.dev "
```

Content of `openblas.pc`:

```
libdir=/workspace/OpenBLAS/install/lib
includedir=/workspace/OpenBLAS/install/include
openblas_config= USE_64BITINT= DYNAMIC_ARCH= DYNAMIC_OLDER= NO_CBLAS= NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP= HASWELL MAX_THREADS=6
version=0.3.12.dev
extralib=-lm -lpthread -lgfortran -lm -lpthread -lgfortran
Name: openblas
Description: OpenBLAS is an optimized BLAS library based on GotoBLAS2 1.13 BSD version
Version: ${version}
URL: https://github.com/xianyi/OpenBLAS
Libs: -L${libdir} -lopenblas
Libs.private: ${extralib}
Cflags: -I${includedir}
```